tests/trainer/test_trainer.py

```diff
@@ -0,0 +1,36 @@
+import unittest
+from unittest.mock import MagicMock
+from torch.utils.data import DataLoader
+from onepiece_classify.data import OnepieceImageDataLoader
+from pathlib import Path
+
+from onepiece_classify.trainer import trainer
+from onepiece_classify.models import image_recog
+
+class TestOnepieceImageDataLoader(unittest.TestCase):
+    def setUp(self):
+        self.root_path = "data"
+        self.batch_size = 32
+        self.num_workers = 4
+
+        self.loader = OnepieceImageDataLoader(
+            self.root_path,
+            self.batch_size,
+            self.num_workers
+        )
+
+        self.model = image_recog(num_classes=18)
+
+        self.training = trainer.fit(self.model, self.loader)
```

Comment:
this is very wrong, do not use you never declare a class Instansiate but you do fit training. and secondly you need to make a separation for testing the fit method, do not write directly inside the constructor setup

```python
def test_trainet_ft(self):
   self.trainer.fit(self.model, self.loader)
   #do your test here
```